### PR TITLE
feature(lxy): add sokoban environment

### PR DESCRIPTION
**Pull request by karroyan**
## TODO
test_sokoban_env.py

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: dizoo/sokoban/envs/sokoban_env.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import gym
+import random
+import marshal
```
Comment (PaParaZz1) [inline dizoo/sokoban/envs/sokoban_env.py:3]: remove unused import

File: dizoo/sokoban/envs/test_sokoban_env.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            print(timestep)
+            print(timestep.obs.max())
+            assert isinstance(timestep.obs, np.ndarray)
+            assert isinstance(timestep.done, bool)
+            assert timestep.obs.shape == (160, 160, 3)
+            print(timestep.info)
+            assert timestep.reward.shape == (1, )
+            assert timestep.reward == -0.1 or timestep.reward == 1.0 or timestep.reward == -1.0 or timestep.reward == 10.0
+            #assert timestep.reward >= env.info().rew_space.value['min']
+            #assert timestep.reward <= env.info().rew_space.value['max']
```
Comment (PaParaZz1) [inline dizoo/sokoban/envs/test_sokoban_env.py:26]: remove commented codes

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: dizoo/sokoban/envs/sokoban_env.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import gym
```
Comment (PaParaZz1) [inline dizoo/sokoban/envs/sokoban_env.py:1]: add this env into [readme env list](https://github.com/opendilab/DI-engine#environment-versatility)

File: dizoo/sokoban/envs/sokoban_env.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            self._env.seed(self._seed)
+        obs = self._env.reset()
+        obs = to_ndarray(obs).astype('float32')
+        print("I add it in sokoban_env", obs)
+        self._final_eval_reward = 0.
+        return obs
+
+    def step(self, action: np.array):
+        action = to_ndarray(action)
+        obs, rew, done, info = self._env.step(int(action))
```
[the remark this answers is not in the document]
Comment (karroyan) [reply dizoo/sokoban/envs/sokoban_env.py:55]: The input of _env.step should be int

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**